src/goesproc/handler_goesr.cc

```diff
@@ -150,6 +173,22 @@ void GOESRImageHandler::handleImage(Tuple t) {
     image->remap(it->second);
   }
 
+  auto grad = config_.gradient.find(details.channel.nameShort);
+  auto idf = imageDataFunction_.begin();
+
+  // If there's a parametric gradient set, use it to generate
+  // an RGB gradient map LUT, then apply it to the image
+  if (grad != std::end(config_.gradient) && idf != imageDataFunction_.end()) {
+    cv::Mat gradientMap(256, 1, CV_8UC3);
+    for (auto i = idf; i != imageDataFunction_.end(); i++) {
+      auto p = grad->second.interpolate(i->second, config_.lerptype);
+      gradientMap.data[i->first * 3] = p.rgb[2] * 255;
+      gradientMap.data[i->first * 3 + 1] = p.rgb[1] * 255;
+      gradientMap.data[i->first * 3 + 2] = p.rgb[0] * 255;
+    }
+    image->remap(gradientMap);
+  }
+
```

Comment:
Can you move this below the LUT handling, then there is no way you can accidentally use both. That will crash hard, so it is good regardless to have an assert or something in the LUT handling that both input images only have 1 channel.